### DISCUSSION TITLE
Possible to join tables from supabase-js rpc function calls?

### DISCUSSION THREAD

**Discussion by davidfant** [supabase/supabase#28717] [Questions]
# Context
I have a Postgres function
```sql
CREATE OR REPLACE FUNCTION public."getTokenLineage"(...)
 RETURNS TABLE("triggerId" uuid, ...)
 LANGUAGE plpgsql
AS $function$
BEGIN
	...
```

This performs a complex filter and returns a list of triggerIds, which map to the `triggers` table. To fetch this I'm doing:

```ts
const triggers = await supabase.rpc("getTokenLineage", { ... });
```

# Problem
When I write the following, supabase-js type inference correctly shows me the right types
```ts
const triggers = await supabase
    .rpc("getTokenLineage", { ... })
    .select(
      `
        triggerId,
        trigger(
          id,
          status
        )
    `
    )
```
<img width="250" alt="image" src="https://github.com/user-attachments/assets/1d9ffdd1-e677-49ec-8aa6-80adac6e0fef">

However that query doesn't work and gives the following error:
```
{
   code: 'PGRST200',
   details: "Searched for a foreign key relationship between 'record' and 'trigger' in the schema 'public', but no matches were found.",
   hint: null,
   message: "Could not find a relationship between 'record' and 'trigger' in the schema cache",
   digest: '2363637366'
}
```

# Question
Is there any way to join relations from rpc calls?

**Comment by GaryAustin1** [comment]
https://docs.postgrest.org/en/v12/references/api/resource_embedding.html#foreign-key-joins-on-table-valued-functions  You can if the returned result is a setof table rows that have a foreign key relation.
![image](https://github.com/user-attachments/assets/4183489e-17bf-413d-b93b-66a7d27df88d)
